crates/services/p2p/src/gossipsub/config.rs

```diff
@@ -54,150 +102,104 @@ pub fn default_gossipsub_builder() -> GossipsubConfigBuilder {
-        ip_colocation_factor_weight: -5.0,
-        ip_colocation_factor_threshold: 10.0,
-        ip_colocation_factor_whitelist: Default::default(),
-
-        ///  Behavioural pattern penalties.
-        ///  This parameter has an associated counter which tracks misbehaviour as detected by the
-        ///  router. The router currently applies penalties for the following behaviors:
-        ///  - attempting to re-graft before the prune backoff time has elapsed.
-        ///  - not following up in IWANT requests for messages advertised with IHAVE.
-        ///
-        ///  The value of the parameter is the square of the counter over the threshold, which decays
-        ///  with BehaviourPenaltyDecay.
-        ///  The weight of the parameter MUST be negative (or zero to disable).
-        behaviour_penalty_weight: -10.0,
-        behaviour_penalty_threshold: 0.0,
-        behaviour_penalty_decay: 0.2,
-
-        /// The decay interval for parameter counters.
-        decay_interval: Duration::from_secs(1),
-
-        /// Counter value below which it is considered 0.
-        decay_to_zero: 0.1,
-
-        /// Time to remember counters for a disconnected peer.
-        retain_score: Duration::from_secs(3600),
-    }
+    let target_value = decay_convergence(
+        params.behaviour_penalty_decay,
+        10.0 / SLOTS_PER_EPOCH as f64,
+    ) - params.behaviour_penalty_threshold;
+
+    params.behaviour_penalty_weight = thresholds.gossip_threshold / target_value.powi(2);
+
+    params.topic_score_cap = MAX_POSITIVE_SCORE * 0.5;
+    params.ip_colocation_factor_weight = -params.topic_score_cap;
+
+    params
 }
 
 fn initialize_peer_score_thresholds() -> PeerScoreThresholds {
     PeerScoreThresholds {
-        /// The score threshold below which gossip propagation is suppressed;
-        /// should be negative.
-        gossip_threshold: -10.0,
-
-        /// The score threshold below which we shouldn't publish when using flood
-        /// publishing (also applies to fanout peers); should be negative and <= `gossip_threshold`.
-        publish_threshold: -50.0,
-
-        /// The score threshold below which message processing is suppressed altogether,
-        /// implementing an effective graylist according to peer score; should be negative and
-        /// <= `publish_threshold`.
-        graylist_threshold: -80.0,
-
-        /// The score threshold below which px will be ignored; this should be positive
-        /// and limited to scores attainable by bootstrappers and other trusted nodes.
-        accept_px_threshold: 10.0,
-
-        /// The median mesh score threshold before triggering opportunistic
-        /// grafting; this should have a small positive value.
-        opportunistic_graft_threshold: 20.0,
+        gossip_threshold: -4000.0,
+        publish_threshold: -8000.0,
+        graylist_threshold: GREYLIST_THRESHOLD,
```

Comment:
Here `graylist_threshold` is spelt with an `a`, but `GREYLIST_THRESHOLD` is spelt with an `e`. Americanos will probably tell you `GRAYLIST` with an `a` is the correct spelling.

Reply:
ah yes I should follow whatever the config is doing, which is surprising since -> https://docs.rs/libp2p/latest/libp2p/swarm/trait.NetworkBehaviour.html 😁